conn/controlfns_linux.go

```diff
@@ -0,0 +1,54 @@
+/* SPDX-License-Identifier: MIT
+ *
+ * Copyright (C) 2017-2022 WireGuard LLC. All Rights Reserved.
+ */
+
+package conn
+
+import (
+	"fmt"
+	"syscall"
+)
+
+func init() {
+	controlFns = append(controlFns,
+
+		// Attempt to set the socket buffer size beyond net.core.{r,w}mem_max by
+		// using SO_*BUFFORCE. This requires CAP_NET_ADMIN, and is allowed here to
+		// fail silently - the result of failure is lower performance on very fast
+		// links or high latency links.
+		func(network, address string, c syscall.RawConn) error {
+			return c.Control(func(fd uintptr) {
+				// Set up to *mem_max
+				_ = syscall.SetsockoptInt(int(fd), syscall.SOL_SOCKET, syscall.SO_RCVBUF, socketBufferSize)
+				_ = syscall.SetsockoptInt(int(fd), syscall.SOL_SOCKET, syscall.SO_SNDBUF, socketBufferSize)
+				// Set beyond *mem_max if CAP_NET_ADMIN
+				_ = syscall.SetsockoptInt(int(fd), syscall.SOL_SOCKET, syscall.SO_RCVBUFFORCE, socketBufferSize)
+				_ = syscall.SetsockoptInt(int(fd), syscall.SOL_SOCKET, syscall.SO_SNDBUFFORCE, socketBufferSize)
+			})
+		},
+
+		// Enable receiving of the packet information (IP_PKTINFO for IPv4,
+		// IPV6_PKTINFO for IPv6) that is used to implement sticky socket support.
+		func(network, address string, c syscall.RawConn) error {
+			var err error
+			switch network {
+			case "udp4":
+				c.Control(func(fd uintptr) {
+					err = syscall.SetsockoptInt(int(fd), syscall.IPPROTO_IP, syscall.IP_PKTINFO, 1)
```

Comment:
Applies everywhere, but I just noticed -- sometimes syscall is used and othertimes unix is used. Should we be centralizing around using the unix package instead? At least for Windows code, syscall is obsolete and x/sys/windows is where it's at.

Reply:
I'll move a bunch over, while it's true the syscall package is mostly frozen, it's also in that regard explicitly more stable, but I don't imagine any of these packages are likely to cause us regressions. As we are going to use things that likely won't arrive in the syscall package it may be more consistent to use the new packages, except where we end up with API/symbol conflicts.

Reply:
Correction: I spoke too soon. There are a bunch of places where we're using syscall because it defines symbols that are portable across unix and windows, for example `syscall.EAFNOSUPPORT` which in a windows build is equivalent to the windows.WSAEAFNOSUPPORT symbol, but removes the need to bifurcate large volumes of code such as about 50% of bind_std.go.

Reply:
So let's use `unix.` where possible, and only use `syscall.` when absolutely necessary? Or, perhaps a better way of thinking of it is to use `unix.` in files that are _unix-tagged, and `windows.` in files that are _windows-tagged, and then `syscall.` is leftover for the shared files.

Does that seem reasonable?

Reply:
James applied this in https://github.com/WireGuard/wireguard-go/pull/64/commits/f54aa5ac9e053b82e806f51e956d185bbcac553c.

Reply:
Looks like `syscall.` is still used in the Windows portion of https://github.com/WireGuard/wireguard-go/commit/aeed2bac0755cc595ffb6f7d248f4c8c2a61b8e7

Reply:
Updated in https://github.com/WireGuard/wireguard-go/commit/a6db822202b983952d475f78819075a40004ca8e